utils/io/content/contentreader.go

```diff
@@ -93,18 +100,25 @@ func (cr *ContentReader) Reset() {
 
 // Cleanup the reader data.
 func (cr *ContentReader) Close() error {
-	if cr.filePath != "" {
-		if err := errorutils.CheckError(os.Remove(cr.filePath)); err != nil {
+	var failedPaths []string
+	for _, filePath := range cr.filesPaths {
+		if filePath == "" {
+			continue
+		}
+		if err := errorutils.CheckError(os.Remove(filePath)); err != nil {
 			log.Error(err)
-			return err
+			failedPaths = append(failedPaths, filePath)
```

Comment:
It looks like we're accumulating this list only for length check at the end.
Can't we just use an int for this?

Reply:
No, because in case some of the files fail to close, then this list becomes the new filesPaths list.